src/lmdb.mli

```diff
@@ -137,7 +131,7 @@ end
 ]}
   *)
   val go :
-    'perm perm ->
+    ?perm:'perm perm ->
```

Comment:
I like the change. I did not manage to get this to type with GADTs.
Problem is that the ``'perm`` type would become polymorphic when Env.t gets monomorphic and therefore there won't be an argument feeding in a ``'perm`` type. :confounded:.
Or is there a way to declare the type of the default argument to an optional argument? (This sounds complicated. You get what I'm asking?)

Reply:
Yes, I do, and no it's not possible to do that ... but, if we use the scheme where `val rw : 'a perm`, then it's ok!

Reply:
Could you elaborate on that scheme? You want types like that?
``` ocaml
type 'a perm =
  | Ro : [ `Read ] perm
  | Rw : 'a perm
```

Reply:
Yeah, that's the most sensible one, see #19

Reply:
When I try to implement this scheme the value restriction triggers and I get weak types on Rw transactions.

Reply:
... but this seems to be no problem because the weak type ``_[< `Read | `Write > `Read ] Txn.t`` won't be narrowed by any function expecting ``[> …]`` open types. :smile:
Interestingly I get the same behaviour with the old scheme of ``| Rw : [ `Read | `Write ]``.

Reply:
In my recent merge I did not include this change because I think the downsides outweigh what you have to gain from an optional argument:
* while the weak return types would work for the time being I fear they might become a problem later.
* it is not obvious what the default behaviour should be (I would opt for read-only, but this won't work with weak types). So making the decision ``Ro``/``Rw`` explicit will make user code unambigous.
* the permission constructor names ``Ro`` and ``Rw`` are very short and concise, so there is not much to gain from making them optional.

If really want them with a label we could do that, but I like them obligatory and unlabelled, so that I may write:
``` ocaml
let env = Env.create Rw "testenv" in
let map =
Txn.go Ro env @@ fun txn ->
Map.(open_existing Nodup ~key:Conv.string ~value:Conv.string) ~txn env
```